### DISCUSSION TITLE
Anybody get Next.js and Supabase Auth working?

### DISCUSSION THREAD

**Discussion by nawhi** [supabase/supabase#12005] [Questions] [answered]
Hello All,
I've been trying for a few days now to get any sort of auth working with Supabase and haven't managed so far.

I followed the instructions for @supabase-nextjs-auth, first in my own repository, then in a simplified repo [here](https://github.com/nawhi/supabase-nextjs-auth).

However, I always end up with the protected route redirecting straight back to the login page.

I have tried:
- adding a '/redirect' page which listens for `onAuthStateChange` and calling `supabase.auth.setSession()` with the result
- adding a '/redirect' page which manually grabs the access token and refresh token from the hash on the redirect after authentication, setting it in a cookie, and reading that cookie in the middleware to generate the supabase client
- adding waits in various places, as I read it might be a race condition

I have not tried:
- a completely manual implementation relying only on the supabase JS client
- using NextAuth.js's community maintained Supabase extension
- using the deprecated `withMiddlewareAuth` function

Just wondered if anybody had encountered this and had a working example in a repository that they would be willing to share.

All best.

**Comment by jeanhdev** [comment]
Same problem here - I cannot find the documentation on what to do after the user clicks on the confirmation email after signup...
Note: I don't want to use the <Auth /> UI component. I just want to implement a simple sign up with email and password with confirmation e-mail.
After clicking on the e-mail, I get an URL with accessToken and refreshToken hashed and have no idea what to do with them. I tried to do setSession but unsuccessful. Any idea ?

**Comment by nawhi** [comment, marked as answer]
For posterity: my issue was caused by an overly restrictive matcher setting on the Next.js middleware. 
It was redirecting the network request that should have loaded the Supabase client-side library to `/login`, so the Supabase code never actually executed, and therefore the auth cookie was never set.
I found this out by looking in the Network tab and observing lots of 307s on my static JS files.
